src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -0,0 +1,195 @@
+import { toAbsoluteUri } from '../Services/NavigationManager';
+
+export class WebAssemblyResourceLoader {
+  private usedCacheKeys: { [key: string]: boolean } = {};
+  private networkLoads: { [name: string]: LoadLogEntry } = {};
+  private cacheLoads: { [name: string]: LoadLogEntry } = {};
+
+  static async initAsync(): Promise<WebAssemblyResourceLoader> {
+    const bootConfigResponse = await fetch('_framework/blazor.boot.json', {
+      method: 'GET',
+      credentials: 'include'
+    });
+
+    const relativeBaseHref = document.baseURI.substring(document.location.origin.length);
+    const cacheName = `blazor-resources-${relativeBaseHref}`;
```

Comment:
Thanks - added comment:

```
    // Define a separate cache for each base href, so we're isolated from any other
    // Blazor application running on the same origin. We need this so that we're free
    // to purge from the cache anything we're not using and don't let it keep growing,
    // since we don't want to be worst offenders for space usage.
```

It's not enough to rely on the browser clearing stuff from its cache when storage space is low. End users short on space (especially on mobiles) do look at the lists of which sites are using the most space and will have a negative opinion of the ones at the top of the list, and are more likely to target those for manual removal from the cache.

Reply:
This comment sounds great! It clarifies the intention. Thanks!